packages/job-worker/src/playout/timeline/rundown.ts

```diff
@@ -240,18 +240,32 @@ function generateCurrentInfinitePieceObjects(
 	if (previousPartInfinites.get(pieceInstance.infinite.infiniteInstanceId)) {
 		groupClasses.push('continues_infinite')
 	}
+	const pieceEnable = getPieceEnableInsidePart(pieceInstance, currentPartInstanceTimings, currentPartGroup.id)
 
 	let nowInParent = currentPartInfo.nowInPart
-	let isAbsoluteInfinitePartGroup = false
 	if (pieceInstance.startedPlayback) {
 		// Make the start time stick
 		infiniteGroup.enable = { start: pieceInstance.startedPlayback }
-		nowInParent = currentTime - pieceInstance.startedPlayback
-		isAbsoluteInfinitePartGroup = true
+		let infiniteGroupStart = pieceInstance.startedPlayback
+
+		// infiniteGroupStart had an actual timestamp inside and pieceEnable.start being a number
+		// means that it expects an offset from it's parent
+		// The infiniteGroupStart is a timestamp of the actual start of the piece controlObj,
+		// which includes the value of `pieceEnable.start` so we need to offset by that value and avoid trimming
+		// the start of the piece group
+		if (typeof pieceEnable.start === 'number' && pieceEnable.start !== null) {
```

Comment:
Is ` && pieceEnable.start !== null` necessary? When will it both be of type number and equal null?

Reply:
Removed static part of the condition.